### PR TITLE
added option keep=False to nlargests/nsmallest

### LINKED ISSUE #16818: Make parameter keep=False keep duplicates for nlargest/nsmallest
#### Code Sample, a copy-pastable example if possible

```python
>>> s = pd.Series([10,9,8,7,7,7,6])
>>> s.nlargest(4)
0    10
1     9
2     8
3     7
dtype: int64
```
#### Problem description
The docstrings list `False` as one of the possible argument values for `keep`.  pandas raises a `ValueError` when attempting to use this parameter. 

#### Expected Output
It would be nice to have nlargest work like this.

```python
>>> s.nlargest(4, keep=False)
0    10
1     9
2     8
3     7
4     7
5     7
```
#### Output of ``pd.show_versions()``

<details>
# Paste the output here pd.show_versions() here
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.1.final.0
python-bits: 64
OS: Darwin
OS-release: 15.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.20.2
pytest: 3.0.7
pip: 9.0.1
setuptools: 35.0.2
Cython: 0.25.2
numpy: 1.13.0
scipy: 0.19.0
xarray: None
IPython: 6.0.0
sphinx: 1.5.5
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: 1.2.0
tables: 3.4.2
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: 2.4.7
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.7.3
bs4: 4.6.0
html5lib: 0.999999999
sqlalchemy: 1.1.9
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: 0.3.0.post
</details>

### PR DESCRIPTION
**Pull request by tdpetrou**
- [x] closes #16818
- [x] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

I made a minor change to `nlargest/nsmallest` to keep all the values of the last `n` when `keep=False`

### CODE REVIEW

**Comment by gfyoung** [comment]
Closed due to decision in #18559.

**Comment by jreback** [comment]
actually this is prob reasonable.

File: pandas/tests/frame/test_analytics.py [hunk 1]
```diff
@@ -2162,6 +2162,21 @@ def test_n_duplicate_index(self, df_duplicates, n, order):
         expected = df.sort_values(order, ascending=False).head(n)
         tm.assert_frame_equal(result, expected)
 
+    def test_keep_false(self):
+        df = pd.DataFrame({'a': [5, 4, 4, 2, 3, 3, 3, 3],
```
Comment (jreback) [inline pandas/tests/frame/test_analytics.py]: can you add the issue number here

Comment (jreback) [reply pandas/tests/frame/test_analytics.py]: make a more informative name for the test

File: pandas/tests/series/test_analytics.py [hunk 2]
```diff
@@ -1865,3 +1865,14 @@ def test_n(self, n):
         result = s.nsmallest(n)
         expected = s.sort_values().head(n)
         assert_series_equal(result, expected)
+
+    def test_keep_false(self):
+        s = Series([10, 9, 8, 7, 7, 7, 7, 6])
```
Comment (jreback) [inline pandas/tests/series/test_analytics.py]: same as above

File: pandas/core/algorithms.py [hunk 3]
```diff
@@ -910,8 +910,8 @@ def __init__(self, obj, n, keep):
         self.n = n
         self.keep = keep
 
-        if self.keep not in ('first', 'last'):
```
Comment (jreback) [inline pandas/core/algorithms.py]: need to add back to the docs-string (don't need a version added as it was tehre before)

**Review by jreback — COMMENTED** [review]
change the whatsnew note which remove the ``keep=False`` (but keep the actual issue number attached there)

**Comment by gfyoung** [comment]
@jreback : What do you think of making `keep=False` be the option?  Besides API consistency, semantically, it doesn't make as much sense to me (or @tdpetrou ) to use because it would imply that we drop ties.

**Comment by tdpetrou** [comment]
@jreback As @gfyoung mentioned, can we choose the parameter value first before making more changes. Should we do 'all', `False`, or something completely different?

**Comment by jreback** [comment]
yeah ``False`` doesn't seem intuitive, ``all`` I think would be ok. Not sure if we have this option anywhere else (e.g. it doesn't exist in ``.drop_duplicates`` nor should it as that would be silly:>)

**Comment by tdpetrou** [comment]
@jreback Changes complete. Also, I didn't see anything in whatsnew on the other issues.

**Comment by jreback** [comment]
pls rebase

**Comment by jreback** [comment]
closing as stale

**Comment by gfyoung** [comment]
@jreback : Revived in #21650.

**Pull request closed without merging**